src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -346,7 +350,8 @@ protected void AddCoins(TransactionBuildContext context)
                 // then it's safe to stop adding UTXOs to the coin list.
                 // The primary goal is to reduce the time it takes to build a trx
                 // when the wallet is bloated with UTXOs.
-                if (sum > totalToSend)
+                // Selected inputs are added first. Ensure that they are all added.
+                if (sum > totalToSend && index >= (context?.SelectedInputs.Count ?? 0))
```

Comment:
difficult read.
How about doing `coins.Add(..)` right at the end of the `if (context.SelectedInputs != null && context.SelectedInputs.Any())` block? Then the rest of the AddCoins method can stay as is.

Reply:
Yes did consider that but it will also require code to prevent coins from being added twice.

Reply:
what you can do is remove them from the list of `context.UnspentOutputs` as you add them, like we do in line 331.

Reply:
that won't work because outputs that will be spent are not expected to be removed from that list and doing so does cause issues albeit not local to this method. i don't think we want to affect the expected behavior of `.AddCoins` unless we are willing to make more extensive changes. I've made the changes that I've described above.